src/frontend/app/components/BrowseScreenSearchWidget.tsx

```diff
@@ -106,43 +116,37 @@ const SearchResults = ({
       }
 
       onKeyphraseChange({ keyphrase: value });
-
-      // Set the input value manually when user clicks on a keyword suggestion
-      if (isSuggestion) {
-        const searchInput = document.getElementById('search-input') as HTMLInputElement;
-        searchInput.value = value;
-      }
     },
     [onKeyphraseChange, router, pathname]
   );
 
-  const onKeyphraseChangeDebounced = debounce((value: string) => handleKeyphraseChange(value), 500);
```

Comment:
It's working fine, however now with debounce gone there are a lot of unnecessary requests. Could we instead add a state variable and update it based on whether or not the input value is empty? And use that variable to show autocomplete or not?